### PR TITLE
Form validity checkbox

### PR DESCRIPTION
**Pull request by aesteves60**

### CODE REVIEW

File: packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx [hunk 1]
```diff
@@ -29,6 +29,11 @@ export class OdsCheckbox {
   @Event() odsFocus!: EventEmitter<void>;
   @Event() odsReset!: EventEmitter<void>;
 
+  @Method()
+  async checkValidity(): Promise<boolean | undefined> {
+    return this.inputEl?.checkValidity();
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:34]: update isInvalid state also?

File: packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx [hunk 2]
```diff
@@ -40,14 +45,24 @@ export class OdsCheckbox {
...
+  }
+
   @Method()
   async getValidity(): Promise<ValidityState | undefined> {
     return this.inputEl?.validity;
   }
 
+  @Method()
+  async reportValidity(): Promise<boolean | undefined> {
+    return this.inputEl?.reportValidity();
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:60]: update isInvalid state also?

File: packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx [hunk 3]
```diff
@@ -80,22 +105,36 @@ export class OdsCheckbox {
...
+            'ods-checkbox__checkbox': true,
+            'ods-checkbox__checkbox--error': this.isInvalid,
+          }}
           checked={ this.isChecked }
           disabled={ this.isDisabled }
           id={ this.inputId }
           indeterminate={ this.isIndeterminate }
           name={ this.name }
-          onBlur={ (): CustomEvent<void> => this.odsBlur.emit() }
+          onInvalid={ (e): void => this.onInvalidEvent(e) }
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:133]: order alphabetically ;)

**Review by dpellier — CHANGES_REQUESTED** [review]

File: packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx [hunk 4]
```diff
@@ -80,22 +107,36 @@ export class OdsCheckbox {
...
+
+    // Enforce the state here as we may still be in pristine state (if the form is submitted before any changes occurs)
+    this.isInvalid = true;
+  }
+
   render(): FunctionalComponent {
     return (
-      <Host class="ods-checkbox">
+      <Host class="ods-checkbox"
+        disabled={ this.isDisabled }>
```
Comment (dpellier) [inline packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:122]: isn't the required also necessary?

Comment (aesteves60) [reply packages/ods/src/components/checkbox/src/components/ods-checkbox/ods-checkbox.tsx:122]: It's the readonly but the checkbox hasn't readonly

**Review by dpellier — APPROVED** [review]

**Pull request merged**